server/cloudapps/form.go

```diff
@@ -0,0 +1,71 @@
+package cloudapps
+
+type Form struct {
+	// RefreshOnChangeTo indicates that changes to the listed fields must reload
+	// the form. Values of the fields with values that are not included in the
+	// refreshed form are lost. Values that no longer apply are reset.
+	RefreshOnChangeTo []string `json:"refresh_on_change_to,omitempty"`
+	RefreshURL        string
+
+	Elements []interface{} // of *XXXElement
+}
+
+type ElementType string
+
+const (
+	ElementTypeCommand       = ElementType("command")
+	ElementTypeText          = ElementType("text")
+	ElementTypeStaticSelect  = ElementType("static_select")
+	ElementTypeDynamicSelect = ElementType("dynamic_select")
+	// ElementTypeTime          = ElementType("time")
+	ElementTypeBool    = ElementType("bool")
+	ElementTypeUser    = ElementType("user")
+	ElementTypeChannel = ElementType("channel")
```

Comment:
We also need a `textarea` type that is already supported by the interactive dialogs

Reply:
I _thought_ it was a subtype or some other flag - as a data type, only `text` makes sense. 2/5 separate presentation from encoding.